content/docs/scaffolder/self-hosted-scaffolder-actions/index.md

```diff
@@ -0,0 +1,152 @@
+
+#### Option 2 - Installing the library manually
+
+You can also install the library manually to your project. The Roadie Agent library can be included in any node.js application and incorporated to existing projects if wanted. The library allows you to run multiple scaffolder action and custom entity providers from within your own infrastructure. To get started, you need to add the node.js library `@roadiehq/roadie-agent` as a dependency into your project and implement your wanted actions or provider using the instructions in the repository. 
+
+### Writing your first action
+
+The agent library is written in Typescript and thus custom actions can be written in either Javascript or using Javascript as a bridge to trigger other commands on the host system. 
+
+To get started, you need to define your configurations on your Roadie instance. This enables the communication between your self-hosted action and the Roadie backend. The underlying communication pattern used for this secure connection is using the [Broker Library](https://github.com/snyk/broker) developed by security company Snyk. You can read more about the broker [in the dedicated documentation section](/docs/integrations/broker/). 
+
+An example custom action, using the repository bootstrapped by the Scaffolder template mentioned above, would look something like below.
+
+First an index file configuring the agent and some actions for it:
+
+```javascript
+// src/index.js
+import { RoadieAgent, createRoadieAgentScaffolderAction } from '@roadiehq/roadie-agent';
+import { addAddtionalFuncToProjectActionHandler } from './addAddtionalFuncToProjectActionHandler.js';
+import { configureMonitoringActionHandler } from './configureMonitoringActionHandler.js';
+
+RoadieAgent.fromConfig({
+  server: 'https://my-tenant.broker.roadie.so',
+  accept: `${process.cwd()}/config/accept.json`,
+  identifier: process.env.BROKER_TOKEN // given that you have exported and env variable called BROKER_TOKEN
+})
+  .addScaffolderAction(createRoadieAgentScaffolderAction({
+    name: 'add-additional-functionality-to-project-action',
+    handler: addAddtionalFuncToProjectActionHandler
+  }))  
+  .addScaffolderAction(createRoadieAgentScaffolderAction({
+    name: 'configure-monitoring-action',
+    handler: configureMonitoringActionHandler
+  }))
+  .start();
+```
+
+An example of an individual action:
+
+```javascript
+// src/addAddtionalFuncToProjectActionHandler.js
+import { writeFileSync } from "fs";
+
+export const addAddtionalFuncToProjectActionHandler = async (context) => {
+  try {
+    const greeting = `Hello, ${JSON.parse(context.payload.body)['name'] || 'world!'}`
+    await context.log(greeting);
+    writeFileSync(`${context.workspacePath}/greeting.txt`, greeting);
```

Comment:
Can a step return output?

Reply:
No, we are currently not mapping responses/outputs back to the action